### ISSUE TITLE
tag wire:id render as text if any comment text html

### ISSUE THREAD

**Issue by maulana-reza** [livewire/livewire#2575]
### Description
wire:id render as text
i forgot remove this comment 
```
<!-- This example requires Tailwind CSS v2.0+ -->
```
and become this
![bug](https://user-images.githubusercontent.com/32976777/109381507-628ee600-790d-11eb-9cfd-e50549fe400d.png)

### Exact steps to reproduce
add this script to show bug
```<!-- This example requires Tailwind CSS v2.0+ -->
<div class="py-12">
    <div class="max-w-7xl mx-auto sm:px-6 lg:px-8">
        <div class="bg-white shadow overflow-hidden sm:rounded-lg">
            <div class="px-4 py-5 sm:px-6">
                <h3 class="text-lg leading-6 font-medium text-gray-900">
                    Applicant Information
                </h3>
                <p class="mt-1 max-w-2xl text-sm text-gray-500">
                    Personal details and application.
                </p>
            </div>
            <div class="border-t border-gray-200">
                <dl>
                    <div class="bg-gray-50 px-4 py-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-6">
                        <dt class="text-sm font-medium text-gray-500">
                            Full name
                        </dt>
                        <dd class="mt-1 text-sm text-gray-900 sm:mt-0 sm:col-span-2">
                            Margot Foster
                        </dd>
                    </div>
                    <div class="bg-white px-4 py-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-6">
                        <dt class="text-sm font-medium text-gray-500">
                            Application for
                        </dt>
                        <dd class="mt-1 text-sm text-gray-900 sm:mt-0 sm:col-span-2">
                            Backend Developer
                        </dd>
                    </div>
                    <div class="bg-gray-50 px-4 py-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-6">
                        <dt class="text-sm font-medium text-gray-500">
                            Email address
                        </dt>
                        <dd class="mt-1 text-sm text-gray-900 sm:mt-0 sm:col-span-2">
                            margotfoster@example.com
                        </dd>
                    </div>
                    <div class="bg-white px-4 py-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-6">
                        <dt class="text-sm font-medium text-gray-500">
                            Salary expectation
                        </dt>
                        <dd class="mt-1 text-sm text-gray-900 sm:mt-0 sm:col-span-2">
                            $120,000
                        </dd>
                    </div>
                    <div class="bg-gray-50 px-4 py-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-6">
                        <dt class="text-sm font-medium text-gray-500">
                            About
                        </dt>
                        <dd class="mt-1 text-sm text-gray-900 sm:mt-0 sm:col-span-2">
                            Fugiat ipsum ipsum deserunt culpa aute sint do nostrud anim incididunt cillum culpa
                            consequat. Excepteur qui ipsum aliquip consequat sint. Sit id mollit nulla mollit nostrud in
                            ea officia proident. Irure nostrud pariatur mollit ad adipisicing reprehenderit deserunt qui
                            eu.
                        </dd>
                    </div>
                    <div class="bg-white px-4 py-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-6">
                        <dt class="text-sm font-medium text-gray-500">
                            Attachments
                        </dt>
                        <dd class="mt-1 text-sm text-gray-900 sm:mt-0 sm:col-span-2">
                            <ul class="border border-gray-200 rounded-md divide-y divide-gray-200">
                                <li class="pl-3 pr-4 py-3 flex items-center justify-between text-sm">
                                    <div class="w-0 flex-1 flex items-center">
                                        <!-- Heroicon name: solid/paper-clip -->
                                        <svg class="flex-shrink-0 h-5 w-5 text-gray-400"
                                             xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor"
                                             aria-hidden="true">
                                            <path fill-rule="evenodd"
                                                  d="M8 4a3 3 0 00-3 3v4a5 5 0 0010 0V7a1 1 0 112 0v4a7 7 0 11-14 0V7a5 5 0 0110 0v4a3 3 0 11-6 0V7a1 1 0 012 0v4a1 1 0 102 0V7a3 3 0 00-3-3z"
                                                  clip-rule="evenodd"/>
                                        </svg>
                                        <span class="ml-2 flex-1 w-0 truncate">
                  resume_back_end_developer.pdf
                </span>
                                    </div>
                                    <div class="ml-4 flex-shrink-0">
                                        <a href="#" class="font-medium text-indigo-600 hover:text-indigo-500">
                                            Download
                                        </a>
                                    </div>
                                </li>
                                <li class="pl-3 pr-4 py-3 flex items-center justify-between text-sm">
                                    <div class="w-0 flex-1 flex items-center">
                                        <!-- Heroicon name: solid/paper-clip -->
                                        <svg class="flex-shrink-0 h-5 w-5 text-gray-400"
                                             xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor"
                                             aria-hidden="true">
                                            <path fill-rule="evenodd"
                                                  d="M8 4a3 3 0 00-3 3v4a5 5 0 0010 0V7a1 1 0 112 0v4a7 7 0 11-14 0V7a5 5 0 0110 0v4a3 3 0 11-6 0V7a1 1 0 012 0v4a1 1 0 102 0V7a3 3 0 00-3-3z"
                                                  clip-rule="evenodd"/>
                                        </svg>
                                        <span class="ml-2 flex-1 w-0 truncate">
                  coverletter_back_end_developer.pdf
                </span>
                                    </div>
                                    <div class="ml-4 flex-shrink-0">
                                        <a href="#" class="font-medium text-indigo-600 hover:text-indigo-500">
                                            Download
                                        </a>
                                    </div>
                                </li>
                            </ul>
                        </dd>
                    </div>
                </dl>
            </div>
        </div>
    </div>
</div>
```
this what i got
```
< wire:id="YZ3to1114MYxdqDON3Xf" wire:initial-data="{"fingerprint":{"id":"YZ3to1114MYxdqDON3Xf","name":"super-admin.pkp-view","locale":"en","path":"super-admin\/pkp\/view\/1","method":"GET"},"effects":{"listeners":[]},"serverMemo":{"children":[],"errors":[],"htmlHash":"0a890a15","data":{"pkp_id":null},"dataMeta":[],"checksum":"063ee834c5c21c3a481088a56fba43263be2462597d9fe2ec4042d1f8c77e6c8"}}"!-- This example requires Tailwind CSS v2.0+ -->
```
**Context**
- Livewire version: [e.g. 2.2.3]
- Laravel version: [e.g. 8.3.1]
- Alpine version: [e.g. 2.3.0]
- Browser: [e.g. Chrome, Safari]

**Comment by austenc** [comment]
Hi @maulana-reza do you think you could provide a repository we could clone to easily reproduce this?

From what you've given here, it's not clear exactly how to reproduce the problem. If you can provide a repository or a more precise example we might be able to help. Thanks!

You may also try upgrading to the latest versions of Livewire and Alpine to see if that fixes your problem.

**Comment by ryangjchandler** [comment]
@austenc I've had this issue before. I'm guessing it's where Livewire is doing a `preg_match` here: https://github.com/livewire/livewire/blob/b647ac63378ea02bdac975f70a4e57aea84bda0b/src/HydrationMiddleware/AddAttributesToRootTagOfHtml.php#L18

This RegEx would actually match the opening `<` on the comment.

**Comment by joshhanley** [comment]
Yeah the issue is the html comment, as html comment is actually a valid DOM element.

You can only have one root element in a Livewire component, which actually includes html comments.

If you want comments, I'd suggest using blade comments `{{-- comment --}}` instead.

**Comment by ryangjchandler** [comment]
@joshhanley How about I throw in a PR to update the RegEx so that it doesn't match comments?

**Comment by joshhanley** [comment]
@ryangjchandler yeah probably a good idea, I don't see any issue with that.

Just need to make sure it skips that whole comment block though. For example if someone has commented out code then you don't want Livewire injecting into one of those commented out elements.

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)

**Comment by orware** [comment]
I just ran into this issue myself while converting a theme over so I could it with Livewire (and the theme had some HTML comments around the section I placed into the Livewire component).

Luckily, I was able to sort out the issue relatively quickly, but others might not so it would be helpful to have an alternative option available (or maybe some that strips out HTML comments before processing? I guess it depends on what's occurring under the hood). The existing knowledge I had that @joshhanley shared above regarding the single root element helped me to sort out the issue before coming here to see if something similar had already been reported.

I guess another option might be to trigger a harder error maybe with an error description if the root element is detected to be an HTML comment, in order to clue people in to the issue in their blade file?
